tests/functional/manage/test_views.py

```diff
@@ -48,6 +53,62 @@ def test_save_account(self, pyramid_services, user_service, db_request):
         assert user.name == "new name"
         assert user.public_email is None
 
+    def test_changing_password_succeeds(self, webtest, remote_addr, socket_enabled):
+        """A user can log in, and change their password."""
+        # create a User
+        user = UserFactory.create(
+            with_verified_primary_email=True, clear_pwd="password"
+        )
+
+        # visit login page
+        login_page = webtest.get("/account/login/", status=HTTPStatus.OK)
+
+        # Fill & submit the login form
+        login_form = login_page.forms[2]  # TODO: form should have an ID, doesn't yet
```

Comment:
Add IDs to the existing forms and replace these with ID lookups

Reply:
https://github.com/pypi/warehouse/pull/16732/commits/17fd99028ee584718e3041d5faf86e1395054dfb